test/qa_scheduler.cpp

```diff
@@ -0,0 +1,222 @@
+#include <boost/ut.hpp>
+
+#include <scheduler.hpp>
+
+#if defined(__clang__) && __clang_major__ >= 16
+// clang 16 does not like ut's default reporter_junit due to some issues with stream buffers and output redirection
+template <>
+auto boost::ut::cfg<boost::ut::override> = boost::ut::runner<boost::ut::reporter<>>{};
+#endif
+
+namespace fg = fair::graph;
+
+using trace_vector = std::vector<std::pair<std::string, std::size_t>>;
+static void trace(trace_vector &traceVector, std::string_view id, std::size_t n = 1) {
+    if (!traceVector.empty() && traceVector.back().first == id) {
+        traceVector.back().second += n;
+    } else {
+        traceVector.emplace_back(id, n);
+    }
+}
+
+template<>
+struct fmt::formatter<std::pair<std::string, std::size_t>> {
+  template <typename ParseContext>
+  constexpr auto parse(ParseContext& ctx) {
+    return ctx.begin();
+  }
+
+  template <typename FormatContext>
+  auto format(const std::pair<std::string, std::size_t> & fp, FormatContext& ctx) {
+    return format_to(ctx.out(), "{} \t{}", fp.first, fp.second);
```

Comment:
please do not use `\ŧ' aligned formatting since this can fail/align more or less randomly... better use, for example, `{:20} {:10}`.